### PR TITLE
MINOR: Rename RaftConfig to QuorumConfig

### PR DESCRIPTION
**Pull request by OmniaGM**
For context https://github.com/apache/kafka/pull/15775#issuecomment-2075340243 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: raft/src/main/java/org/apache/kafka/raft/QuorumConfigs.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
  * controller should be able to transition from standby to active without reloading all of
  * the metadata. The standby is a "hot" standby, not a "cold" one.
  */
-public class RaftConfig {
+public class QuorumConfigs {
```
Comment (chia7712) [inline raft/src/main/java/org/apache/kafka/raft/QuorumConfigs.java]: It seems to me QuorumConfig`s` implies that we should not have instance of it. However, we do instantiate it, and so could we rename it to `QuorumConfig`? Another similar case is `LogConfig`.

**Comment by chia7712** [comment]
@OmniaGM Could you please rebase code to include fix (https://github.com/apache/kafka/pull/15801)?

**Comment by OmniaGM** [comment]
> @OmniaGM Could you please rebase code to include fix (#15801)?

done

**Review by chia7712 — APPROVED** [review]
LGTM

**Pull request merged**